src/lib/core/components/visualizations/VisualizationsContainer.tsx

```diff
@@ -130,21 +130,34 @@ function ConfiguredVisualizations(props: Props) {
                       </button>
                     </Tooltip>
                   </div>
-                  <div>
-                    <Tooltip title="View fullscreen">
-                      <Link replace to={`${url}/${viz.visualizationId}`}>
-                        <i className="fa fa-arrows-alt"></i>
-                      </Link>
-                    </Tooltip>
-                  </div>
                 </div>
-                {viz.descriptor.thumbnail ? (
-                  <img alt={viz.displayName} src={viz.descriptor.thumbnail} />
-                ) : (
-                  <div className={cx('-ConfiguredVisualizationNoPreview')}>
-                    Preview unavaiable
-                  </div>
-                )}
+                {/* add the Link of thumbnail box here to avoid click conflict with icons */}
+                <>
+                  <Link replace to={`${url}/${viz.visualizationId}`}>
+                    {viz.descriptor.thumbnail ? (
+                      <img
+                        alt={viz.displayName}
+                        src={viz.descriptor.thumbnail}
+                      />
+                    ) : (
+                      <div className={cx('-ConfiguredVisualizationNoPreview')}>
+                        Preview unavaiable
+                      </div>
+                    )}
+                    {/* make gray-out box on top of thumbnail */}
+                    {!arraysEqual(
+                      props.filters,
+                      (viz.descriptor.configuration as any).currentPlotFilters
+                    ) ? (
+                      <div className={cx('-ConfiguredVisualizationGrayOut')}>
+                        Open to sync with
+                        <br /> current subset
+                      </div>
+                    ) : (
+                      ''
+                    )}
+                  </Link>
+                </>
```

Comment:
@jtlong3rd I have changed custom functions to Lodash's isEqual. Yes I know that it would be expensive so your idea is good to go. I have complied with 👍